internal/reconciler/logpipeline/otel/reconciler.go

```diff
@@ -301,6 +303,46 @@ func (r *Reconciler) getReplicaCountFromTelemetry(ctx context.Context) int32 {
 	return defaultReplicaCount
 }
 
+func (r *Reconciler) getPresetsFromTelemetry(ctx context.Context) gatewayprocs.Presets {
+	var telemetries operatorv1alpha1.TelemetryList
+	if err := r.List(ctx, &telemetries); err != nil {
+		logf.FromContext(ctx).V(1).Error(err, "Failed to list telemetry: using default presets")
+		return gatewayprocs.Presets{}
+	}
+
+	for i := range telemetries.Items {
+		telemetrySpec := telemetries.Items[i].Spec
+		if telemetrySpec.Log == nil {
+			continue
+		}
+
+		if telemetrySpec.Log.Presets == nil {
+			continue
+		}
+
+		mapPodLabels := func(values []operatorv1alpha1.PodLabel, fn func(operatorv1alpha1.PodLabel) gatewayprocs.PodLabel) []gatewayprocs.PodLabel {
+			var result []gatewayprocs.PodLabel
+			for i := range values {
+				result = append(result, fn(values[i]))
+			}
+
+			return result
+		}
+
+		return gatewayprocs.Presets{
+			Enabled: telemetrySpec.Log.Presets.Enabled,
+			PodLabels: mapPodLabels(telemetrySpec.Log.Presets.PodLabels, func(value operatorv1alpha1.PodLabel) gatewayprocs.PodLabel {
+				return gatewayprocs.PodLabel{
+					Key:       value.Key,
+					KeyPrefix: value.KeyPrefix,
+				}
+			}),
+		}
+	}
```

Comment:
So, in case of multiple telemetries you will take the first one? Is it what we want?

Reply:
Refactored for all places relaying on fist found instance, the new code will get the default instance from manager namespace